### ISSUE TITLE
pygame.Rect.__getitem__, __setitem__ and __delitem__ error message inconsistency

### ISSUE THREAD

**Issue by algorithmpub** [pygame/pygame#1251]
**System info:**
pygame 1.9.3 on python 3.5.2 on 32 bit Lubuntu (Ubuntu 16.04.6 LTS).
pygame installed via "sudo pip3 install pygame" (pip3 is pip 18.0, python 3.5)

**Issue:**
As can be seen in the example below, `pygame.Rect.__getitem__, __setitem__ and __delitem__` present different error messages for the same operations when performed with different syntaxes.

```
>>> from pygame import Rect
>>> r = Rect(10, 20, 30, 40)

>>> r[2.2]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence index must be integer, not 'float'
>>> r.__getitem__(2.2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'float' object cannot be interpreted as an integer

>>> r[2.2] = 20
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence index must be integer, not 'float'
>>> r.__setitem__(2.2, 20)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'float' object cannot be interpreted as an integer

>>> del r[2.2]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence index must be integer, not 'float'
>>> r.__delitem__(2.2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'float' object cannot be interpreted as an integer

```

I'm not sure if one of the recent pull requests regarding the pygame.Rect class already fixed this issue, but since I couldn't find anything related to this, I decided to report. If this was already fixed, please disregard and close.

As always, thanks for the selfless dedication of your time and effort.

Edit: added other methods which were also misbehaving.

**Comment by charlesej** [comment]
Thanks for the issue. This has been fixed (see 703350f68182641fe72ef1f41d81998fb6d4b385).

```python
import pygame

rect = pygame.Rect(1, 1, 1, 1)
idx = 1.1

print('\nget')
try:
    rect[idx]
except TypeError as e:
    print(e)

try:
    rect.__getitem__(idx)
except TypeError as e:
    print(e)


print('\nset')
try:
    rect[idx] = 2
except TypeError as e:
    print(e)

try:
    rect.__setitem__(idx, 2)
except TypeError as e:
    print(e)


print('\ndel')
try:
    del rect[idx]
except TypeError as e:
    print(e)

try:
    rect.__delitem__(idx)
except TypeError as e:
    print(e)
```

---

```
pygame 2.0.0.dev3 (SDL 2.0.10, python 3.7.2)
Hello from the pygame community. https://www.pygame.org/contribute.html

get
Invalid Rect slice
Invalid Rect slice

set
Invalid Rect slice
Invalid Rect slice

del
Invalid Rect slice
Invalid Rect slice
```

---

```
pygame 2.0.0.dev3 (SDL 2.0.10, python 2.7.10)
Hello from the pygame community. https://www.pygame.org/contribute.html

get
Invalid Rect slice
Invalid Rect slice

set
Invalid Rect slice
Invalid Rect slice

del
Invalid Rect slice
Invalid Rect slice
```

**Comment by algorithmpub** [comment]
This are great news. Thanks.
